src/internal/connector/graph/uploadsession.go

```diff
@@ -48,17 +48,25 @@ func (iw *writer) Write(p []byte) (int, error) {
 
 	// PUT the request - set headers `Content-Range`to describe total size and `Content-Length` to describe size of
 	// data in the current request
-	_, err := iw.client.R().
-		SetHeaders(map[string]string{
-			contentRangeHeaderKey: fmt.Sprintf(
-				contentRangeHeaderValueFmt,
-				iw.lastWrittenOffset,
-				endOffset-1,
-				iw.contentLength),
-			contentLengthHeaderKey: fmt.Sprintf("%d", rangeLength),
-		}).
-		SetBody(bytes.NewReader(p)).
-		Put(iw.url)
+	req, err := http.NewRequest("PUT", iw.url, bytes.NewReader(p))
+	if err != nil {
+		return 0, clues.Wrap(err, "uploading item").With(
```

Comment:
These `with` values seem useful for the whole call chain.  Can you pull them out to a `clues.Add(ctx...)` above the request instead, please?

Reply:
With refactoring this is not required now.